tox.ini

```
@@ -1,21 +1,17 @@
 [tox]
 isolated_build = True
 envlist =
-    py{38,39,310}-dj32
-    py{38,39,310}-dj40
-    py{38,39,310,311}-dj41
     py{38,39,310,311,312}-dj42
     py{310,311,312}-dj50
+    py{310,311,312}-dj51
     py{310,311,312}-djmain
 
 [testenv]
 pip_pre = True
 deps =
-    dj32: django~=3.2.0
-    dj40: django~=4.0.0
-    dj41: django~=4.1.0
     dj42: django~=4.2.0
     dj50: django~=5.0.0
+    dj51: django~=5.1.0
```

```suggestion
    dj51: django>=5.1,<5.2
```

This might work to clean up the build errors, although to my reading of [the docs](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-matching), it seems to me that this really should work.

---

Bummer, it didn't work. Sorry for the bad suggestion.